### ISSUE TITLE
TypeError: object is not a function

### ISSUE THREAD

**Issue by abilng** [webpack-contrib/compression-webpack-plugin#24]
When I added `new CompressionPlugin()` to plugin array. I am getting a error.

```
zlib.js:167
      callback(null, buf);
      ^
  TypeError: object is not a function
      at Gzip.onEnd (zlib.js:167:5)
      at Gzip.emit (events.js:117:20)
      at _stream_readable.js:944:16
      at process._tickCallback (node.js:448:13)

```

node: v0.10.40
npm: 1.4.28

**Comment by h4l** [comment]
Which version of `compression-webpack-plugin` are you using and which options are you passing?

**Comment by abilng** [comment]
version: "0.3.1"

**Comment by h4l** [comment]
Are you providing your own function with the `algorithm` option like this?:

```
module.exports = {
    context: __dirname + "/app",
    entry: "./entry",
    output: {
        path: __dirname + "/dist",
        filename: "bundle.js"
    },
    plugins: [
        new CompressionPlugin({
            algorithm: function(buf, cb) {
                zlib.gzip(buf, {}, cb);
            }
        })
    ]
};
```

If so the reason is that 0.3.1 changed the arguments of the algorithm function. See #25. You need to change the signature to `function(buf, options, cb)`

**Comment by smitchell** [comment]
I get this error too, on my home machine, but not at work with the same project. It must some version that is different between environments.

"If you do, this is most likely a problem with the angular2-webpack-starter package,"

`zlib.js:227
    callback(err, buf);
    ^

TypeError: callback is not a function
    at Gzip.onEnd (zlib.js:227:5)
    at emitNone (events.js:85:20)
    at Gzip.emit (events.js:179:7)
    at endReadableNT (_stream_readable.js:913:12)
    at _combinedTickCallback (node.js:383:13)
    at process._tickCallback (node.js:407:11)

$ npm -v compression-webpack-plugin
3.8.3
$ node --version
v5.9.0
$ npm -version
3.8.3

**Comment by weltraumpirat** [comment]
I can confirm a signature change will fix the problem with angular2-webpack-starter package. You need to edit `helper.js` from this:

```
function gzipMaxLevel(buffer, callback) {
  return zlib['gzip'](buffer, {level: 9}, callback)
}
```
to this:

```
function gzipMaxLevel(buffer, option, callback) {
  return zlib['gzip'](buffer, {level: 9}, callback)
}
```

**Comment by smitchell** [comment]
Thanks, I made this change on the machine where "build:prod" was failing and verified it was fixed. What library version caused the need for this change? Is this a breaking change for the machines that were working? I assuming I'll need to bring all our machines to some minimum version of that library.

**Comment by weltraumpirat** [comment]
As @h4l said: 
> If so the reason is that 0.3.1 changed the arguments of the algorithm function. See #25. You need to change the signature to function(buf, options, cb)

So 0.3.1 of compression-webpack-plugin changed the API that the angular2-webpack-starter package relies on. The thing is, there's already an open issue that requests the change to be reversed, or at least optional (sic). I probably wouldn't go nuts changing that call on all your machines, if you can postpone it.

**Comment by tbrisker** [comment]
This error also occurs when using compression-webpack-plugin with node 0.10, which doesn't support the options parameter for gzip. 
As a workaround I used the following code:
```javascript
plugins: [
    new CompressionPlugin({
      algorithm: function(buffer, options, callback){
                   zlib.gzip(buffer, callback);
                 }
    })
]
```

**Comment by michael-ciniawsky** [comment]
Will be fixed by either #26 , #27 discussion moved to the respective PR's, feedback highly welcome 😛 . Vote with 👍 / 👎  on the opening post of the PR you prefer
